### ISSUE TITLE
Notifications not working as expected

### ISSUE THREAD

**Issue by EmiliaBlasten** [Glitchy-Tozier/githo#21]
The notification just buzzes my phone and does not leave any visible message in the notification tab. If I miss the buzz there is no way for me to see that I was supposed to do something unless I explicitly open the app.

I have notifications enabled:
![Screenshot_20220413-112128](https://user-images.githubusercontent.com/46397121/163202929-c46f149c-ef12-4b8a-9cd2-d4145f94e040.png)
![Screenshot_20220413-112142](https://user-images.githubusercontent.com/46397121/163202972-84b39ae8-c5c7-43d8-a2fa-5a7f3f32fa79.png)

The Android notifications tab does not display anything related to the app:
![Screenshot_20220413-112313](https://user-images.githubusercontent.com/46397121/163203006-cd17320d-7ef2-430e-8ab1-bdaa7eebf2b6.png)

I am running version 1.3.2 from F-Droid.

edit: the "rotating arrow" notification icon in the first two screenshots is from another app.

_Originally posted by @EmiliaBlasten in https://github.com/Glitchy-Tozier/githo/issues/17#issuecomment-1098120960_

**Comment by Glitchy-Tozier** [comment]
Interesting, thank you for your bug-report!
I was having an issue where notifications wouldn't show up at all; most of the time my phone has its sounds&vibrations disabled though, so it might actually be facing the same bug.

A few questions:
1. Is the notification always on time?
2. Did the notification-time ever reset to 00:00, 02:00 or 23:59 without you doing anything?
3. Have the notifications ever showed up in any scenario or has it always been buzz-only?
4. If you stare at your phone at 11:20, can you see the notification appearing and then dissappearing immediately or does it never even show up at all?

**Comment by EmiliaBlasten** [comment]
Thanks for the quick reply! I have not had the app for long yet, so take my answers with a grain of salt:

>     1. Is the notification always on time?
It has been on time: I set it to 11:20 at 09:00 this morning and it buzzed at the right time. I also tried setting it a few times for a couple of minutes later (e.g. at 11:35 I would set it to 11:37 for testing) and in each case the phone buzzed at the right time.

>     2. Did the notification-time ever reset to 00:00, 02:00 or 23:59 without you doing anything?
The notification time has never changed by itself in the total 20 hours of having it installed.
 
>     3. Have the notifications ever showed up in any scenario or has it always been buzz-only?
So far it has been a buzz only. However I have used the app for less than 24 hours. I am wondering if the weird hours selector might cause the notification to display every other day, but I need to have longer usage before making any guesses.

Weird in-app hours selector:
![Screenshot_20220413-201652](https://user-images.githubusercontent.com/46397121/163285151-bfe0272b-c34c-43df-ab89-2c489a71c598.png)
Compared to the usual Android hour selector:
![Screenshot_20220413-201937](https://user-images.githubusercontent.com/46397121/163285253-4d727fae-26e8-49ac-8247-2df04c4cd009.png)

>     4. If you stare at your phone at 11:20, can you see the notification appearing and then dissappearing immediately or does it never even show up at all?
I don't see a notification appearing even for a split second. So it's only the buzz as far as I can tell.

**Comment by Glitchy-Tozier** [comment]
> I am wondering if the weird hours selector might cause the notification to display every other day, but I need to have longer usage before making any guesses.

1. I agree that it's kind of weird, but it's an official widget, coming directly from Google.
The design is pretty new. In case you're imterested, here's a small article about it: https://www.androidheadlines.com/2021/08/material-you-design-android-time-picker.html
2. The time-picker works flawlessly, something else is broken.

> Everything else

Thank you for providing the information!
The weird thing is, it's difficult to test this bug consistently.
- On your device, there's only buzzing and nothing else
- On my device, there's nothing at all, not even a buzz
- When using my test setup, notifications show up, but sometimes seem appear at the wrong time.

I'll investigate further into this, but it might take me a few days.

**Comment by Glitchy-Tozier** [comment]
I think I managed to fix this issue in [this commit](https://github.com/Glitchy-Tozier/githo/commit/f42251048f97ecc3f05c9d37934cc35a186c4659). The problem was that the notification's icon was removed by the compiler, which caused notifications to break/fail. See [here](https://pub.dev/packages/flutter_local_notifications#release-build-configuration) for more details.

I'll release the next version of Githo today or tomorrow, meaning that the update should show up within 3-5 days on F-droid.
